specification/botservice/resource-manager/Microsoft.BotService/preview/2017-12-01/botservice.json

```diff
@@ -0,0 +1,1809 @@
+        ],
+        "responses": {
+          "200": {
+            "description": "A 200 (OK) should be returned if the object exists and was deleted successfully;"
+          },
+          "204": {
+            "description": "a 204 (NoContent) should be used if the resource does not exist and the request is well formed."
+          },
+          "default": {
+            "description": "Error response describing why the operation failed",
+            "schema": {
+              "$ref": "#/definitions/Error"
+            }
+          }
+        }
+      },
+      "get": {
+        "tags": [
+          "Channel"
+        ],
+        "description": "Returns a BotService Channel registration specified by the parameters.",
+        "operationId": "Channels_Get",
+        "x-ms-examples": {
+          "Get Bot": {
+            "$ref": "./examples/GetChannel.json"
+          }
+        },
+        "parameters": [
+          {
+            "name": "resourceGroupName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The name of the resource group within the user's subscription."
+          },
+          {
+            "name": "resourceName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "maxLength": 64,
+            "minLength": 2,
+            "pattern": "^[a-zA-Z0-9][a-zA-Z0-9_.-]*$",
+            "description": "The name of the Bot resource."
+          },
+          {
+            "name": "channelName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "maxLength": 64,
+            "minLength": 2,
+            "pattern": "^[a-zA-Z0-9][a-zA-Z0-9_.-]*$",
+            "description": "The name of the Bot resource."
+          },
+          {
+            "$ref": "#/parameters/apiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/subscriptionIdParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "The resource provider should return 200 (OK) to indicate that the operation completed successfully. ",
+            "schema": {
+              "$ref": "#/definitions/BotChannel"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed. If the resource group *or* resource does not exist, 404 (NotFound) should be returned.",
+            "schema": {
+              "$ref": "#/definitions/Error"
+            }
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.BotService/botServices/{resourceName}/channels/{channelName}/listChannelWithKeys": {
+      "post": {
+        "tags": [
+          "Channel"
+        ],
+        "description": "Lists a Channel registration for a Bot Service including secrets",
+        "operationId": "Channel_List",

```

Comment:
This should be updated to `"operationId": "Channels_ListWithKeys"`.

Updating "Channel" to "Channels" is to be consistent with the operations above. Without the change, our code generators would generate both a `ChannelClient` and a `ChannelsClient`.

Our RPC bot identified the omitted "WithKeys". The logic being, if it was important enough to convey the "WithKeys" in the path, it must be important enough to convey in the generated SDKs.

Reply:
There's another RPC Violation noted by the bot:
>"The child tracked resource, 'channels' with immediate parent 'Bot', must have a list by immediate parent operation."

Truthfully, I'm not familiar with this error. I believe it's complaining that there isn't an operation registered under the path `"/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.BotService/botServices/{resourceName}/channels/{channelName}/list"`

That could reverse the conclusion of the comment above, and instead of changing the `operationId` to "Channels_ListWithKeys" you may update it to just "Channels_List", but change the path its registered.

Another option is that your service may needs both a "List" and a "ListWithKeys".

If none of those options make sense, we may be able to just add a suppression. Just let me know.